go/mysql/encoding.go

```diff
@@ -332,6 +332,53 @@ func readLenEncStringAsBytesCopy(data []byte, pos int) ([]byte, int, bool) {
 	return result, pos + s, true
 }
 
+// > encGtidData("xxx")
+//
+//	[07 03 05 00 03 78 78 78]
+//	 |  |  |  |  |  |------|
+//	 |  |  |  |  |  ^-------- "xxx"
+//	 |  |  |  |  ^------------ length of rest of bytes, 3
+//	 |  |  |  ^--------------- fixed 0x00
+//	 |  |  ^------------------ length of rest of bytes, 5
+//	 |  ^--------------------- fixed 0x03 (SESSION_TRACK_GTIDS)
+//	 ^------------------------ length of rest of bytes, 7
+//
+// This is ultimately lenencoded strings of length encoded strings, or:
+// > lenenc(0x03 + lenenc(0x00 + lenenc(data)))
+func encGtidData(data string) []byte {
+	const SESSION_TRACK_GTIDS = 0x03
```

Comment:
The project does not use screaming snake case for `consts`, but instead camelCase or PascalCase (depending on if it should be exported or not).

Personal preference, but it's also more standard in this code base to include your consts together near the top of the file. That's very nitty though, so up to you.

Reply:
Ah, well change. I took this from MySQL's [`enum enum_session_state_type`](https://dev.mysql.com/doc/dev/mysql-server/latest/mysql__com_8h.html#a1c6cf2629b0bda6da6788a25725d6b7f) and didn't feel it was relevant to expose outside this function, I just wanted to put a name over the magic `0x03` byte so it made more logical sense.

I'll rename it tho to `SessionTrackGtids`.